apps/trading/lib/hooks/use-navigate-to-last-market.spec.ts

```diff
@@ -0,0 +1,60 @@
+import * as router from 'react-router';
+import { useNavigateToLastMarket } from './use-navigate-to-last-market';
+import { useGlobalStore } from '../../stores';
+import { renderHook } from '@testing-library/react';
+import { useTopTradedMarkets } from './use-top-traded-markets';
+import { Links } from '../links';
+
+const mockLastMarketId = 'LAST';
+
+jest.mock('../../stores', () => {
+  const original = jest.requireActual('../../stores');
+  return {
+    ...original,
+    useGlobalStore: jest.fn(),
+  };
+});
+
+jest.mock('./use-top-traded-markets', () => {
+  return {
+    useTopTradedMarkets: jest.fn(),
+  };
+});
+
+describe('useNavigateToLastMarket', () => {
+  const navigate = jest.fn();
+  beforeAll(() => {
+    jest.spyOn(router, 'useNavigate').mockImplementation(() => navigate);
+  });
+
+  it('navigates to the last market when it is active', () => {
+    (useGlobalStore as unknown as jest.Mock).mockReturnValue(mockLastMarketId);
```

Comment:
I've been enjoying just calling setState directly before rendering for some tests. Feels simpler as you can avoid setting up all the mocks above (also useGlobalStore, gets inherently tested too )

You do need to be careful about the zustand store state leaking between tests though

useGlobalStore.setState({ marketId: mockLastMarketId })